tests/cpp/test_multidevice_matmul.cpp

```diff
@@ -404,4 +404,167 @@ TEST_F(DistributedMatmulTest, Matmul_LayoutNT_ReduceScatter) {
                                     ->heuristic();
   EXPECT_EQ(heuristic, ScheduleHeuristic::ExprEval);
 }
+
+TEST_F(DistributedMatmulTest, MLP_Layer) {
+  std::unique_ptr<Fusion> fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  auto mesh = DeviceMesh::createForNumDevices(communicator_->size());
+
+  int64_t sb = 64; // sequence * batch
+  int64_t h = 128;
+  int64_t h4 = 4 * h;
+
+  // TODO: error with dynamic shape
```

Comment:
Can you clarify this? Are you saying the following code would fail if changed to makeContigTensor?

Reply:
I think it's another instance of #2462. Please revisit when it's fixed.

Reply:
We are using FusionExecutorCache so sadly that did not fix the error. I did narrow down what is causing the error and opened an issue https://github.com/NVIDIA/Fuser/issues/2481